src/scep/Client/client.py

```diff
@@ -114,12 +114,12 @@ def get_crl(self, identity, identity_private_key, serial_number, identifier=None
 
         return self._pki_operation(identity=identity, identity_private_key=identity_private_key, envelope=envelope, message_type=MessageType.GetCRL, cacaps=cacaps, ca_certs=ca_certs)
 
-    def enrol(self, csr, identity, identity_private_key, identifier=None):
+    def enrol(self, csr, identity, identity_private_key, identifier=None, keyEncAlg=None):
         """Perform a PKCSReq operation by submitting a CSR to the SCEP service."""
         cacaps = self.get_ca_capabilities(identifier=identifier)
         ca_certs = self.get_ca_certs(identifier=identifier)
-        envelope = PKCSPKIEnvelopeBuilder().encrypt(csr.to_der(), cacaps.strongest_cipher())
-        transaction_id = hex_digest_for_data(data=csr.public_key.to_der(), algorithm='sha1')
+        envelope = PKCSPKIEnvelopeBuilder().encrypt(csr.to_der(), cacaps.strongest_cipher(), keyEncAlg)
+        transaction_id = hex_digest_for_data(data=csr.public_key.to_der(), algorithm='sha256')
```

Comment:
We are generating sha1 of public key to create a transaction id, sha256 is a longer transaction id and older systems might now support it.